nats.go

```diff
@@ -1208,6 +1227,16 @@ func (nc *Conn) checkForSecure() error {
 		o.Secure = true
 	}
 
+	if o.Nkey != "" && nc.info.Nonce == "" {
+		return ErrNkeysNoSupported
+	}
+
+	// Check if we have an nkey but no signature callback
+	// defined.
+	if o.Nkey != "" && o.SignatureCB == nil {
```

Comment:
Why put this test in here? Since we have a test above that basically says that a client that has NKey cannot connect to a server that does not support it, why not do this test early in Connect()?

Reply:
I can move it..